hail/src/main/scala/is/hail/io/vcf/ExportVCF.scala

```diff
@@ -167,28 +167,47 @@ object ExportVCF {
     }
   }
   
-  def emitGenotype(sb: StringBuilder, formatFieldOrder: Array[Int], tg: PStruct, m: Region, offset: Long) {
-    formatFieldOrder.foreachBetween { j =>
-      val fIsDefined = tg.isFieldDefined(m, offset, j)
-      val fOffset = tg.loadField(m, offset, j)
-
-      tg.fields(j).typ match {
-        case it: PContainer =>
-          val pt = it
-          if (fIsDefined) {
-            val fLength = pt.loadLength(m, fOffset)
-            iterableVCF(sb, pt, m, fLength, fOffset, ',')
-          } else
-            sb += '.'
-        case t =>
-          if (fIsDefined)
-            strVCF(sb, t, m, fOffset)
-          else if (t.virtualType.isOfType(TCall()))
-            sb.append("./.")
-          else
-            sb += '.'
+  def emitGenotype(sb: StringBuilder, formatFieldOrder: Array[Int], tg: PStruct, m: Region, offset: Long, missingnessArray: Array[Boolean]) {
+    var i = 0
+    while (i < formatFieldOrder.length) {
+      missingnessArray(i) = tg.isFieldDefined(m, offset, formatFieldOrder(i))
```

Comment:
Can you add a `!tg.isFieldDefined`. Then it will be true if missing which matches the name of the array. Otherwise, it makes my head hurt a bit.

Reply:
renamed to `fieldDefined`.